packages/dynamite/dynamite/lib/src/models/json_schema/schema.dart

```diff
@@ -146,3 +339,63 @@ class SchemaPlugin implements SerializerPlugin {
   @override
   Object? beforeSerialize(Object? object, FullType specifiedType) => object;
 }
+
+class _JsonSchemaSerializer extends StructuredSerializer<JsonSchema> {
```

Comment:
I see no handling for `$ref`, will all those go into EmptySchema because they have no `type` set?

Reply:
Ah yes this seems to be the case according to the new tests. TBH I find that a bit confusing because I would expect EmptySchema to be equivalent to a `{}` schema and nothing else.

Reply:
Yeah I'm also not a fan.

What about renaming the `JsonSchema` into `BaseSchema` and `EmptySchema` to `JsonSchema`?

Reply:
To me that also feels wrong as I'd expect all other schemas to then extend JsonSchema which they don't. The EmptySchema is only necessary to be able to distinguish it from the other schemas, right? Is that even necessary anywhere?
My idea: Rename EmptySchema to GenericSchema which at least somewhat indicates that it can also have some values but it is not exactly decided which ones they are.